packages/kit/src/runtime/server/index.js

```diff
@@ -143,12 +143,28 @@ export async function respond(request, options, state) {
 				const lower = key.toLowerCase();
 
 				if (lower in headers) {
-					throw new Error(`"${key}" header is already set`);
+					if (lower === 'set-cookie') {
+						if (!Array.isArray(headers[lower])) {
+							headers[lower] = [/**@type{string} */ (headers[lower])];
+						}
+						const cookies = /**@type{string[]} */ (headers[lower]);
+						const new_cookies = /**@type{string[]} */ (
```

Comment:
elsewhere we've been formatting like so:
```suggestion
							headers[lower] = [/** @type{string} */ (headers[lower])];
						}
						const cookies = /** @type{string[]} */ (headers[lower]);
						const new_cookies = /** @type{string[]} */ (
```